From e31ad049ab2c0d0f1ba64bf5b8aa0d3c388a69e4 Mon Sep 17 00:00:00 2001
From: Arthur Williams <taaparthur@gmail.com>
Date: Sun, 27 Mar 2022 03:10:25 -0700
Subject: [PATCH 3/8] dmenu-initialtext-4.7.diff

---
 dmenu.1 | 5 +++++
 dmenu.c | 8 ++++++--
 2 files changed, 11 insertions(+), 2 deletions(-)

diff --git a/dmenu.1 b/dmenu.1
index 57ea184..affdb16 100644
--- a/dmenu.1
+++ b/dmenu.1
@@ -24,6 +24,8 @@ dmenu \- dynamic menu
 .IR color ]
 .RB [ \-w
 .IR windowid ]
+.RB [ \-it
+.IR text ]
 .P
 .BR dmenu_run " ..."
 .SH DESCRIPTION
@@ -88,6 +90,9 @@ prints version information to stdout, then exits.
 .TP
 .BI \-w " windowid"
 embed into windowid.
+.TP
+.BI \-it " text".
+start with initial text input.
 .SH USAGE
 dmenu is completely controlled by the keyboard.  Items are selected using the
 arrow keys, page up, page down, home, and end.
diff --git a/dmenu.c b/dmenu.c
index 552b1fd..04c2d34 100644
--- a/dmenu.c
+++ b/dmenu.c
@@ -737,7 +737,8 @@ static void
 usage(void)
 {
 	fputs("usage: dmenu [-bfiv] [-l lines] [-p prompt] [-fn font] [-m monitor]\n"
-	      "             [-nb color] [-nf color] [-sb color] [-sf color] [-w windowid]\n", stderr);
+	      "             [-nb color] [-nf color] [-sb color] [-sf color] [-w windowid]\n"
+	      "             [-it text]\n", stderr);
 	exit(1);
 }
 
@@ -786,7 +787,10 @@ main(int argc, char *argv[])
 			colors[SchemeSel][ColFg] = argv[++i];
 		else if (!strcmp(argv[i], "-w"))   /* embedding window id */
 			embed = argv[++i];
-		else
+		else if (!strcmp(argv[i], "-it")) {   /* embedding window id */
+			const char * text = argv[++i];
+			insert(text, strlen(text));
+		} else
 			usage();
 
 	if (!setlocale(LC_CTYPE, "") || !XSupportsLocale())
-- 
2.35.1

