From be16a4568fa8dd3cacf18f609cf408132786be5a Mon Sep 17 00:00:00 2001
From: Arthur Williams <taaparthur@gmail.com>
Date: Sat, 20 Mar 2021 18:35:30 -0700
Subject: [PATCH 5/8] Add keybindings for lower/raise/power buttons

---
 dmenu.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/dmenu.c b/dmenu.c
index 096846b..3e06b95 100644
--- a/dmenu.c
+++ b/dmenu.c
@@ -1,4 +1,5 @@
 /* See LICENSE file for copyright and license details. */
+#include <X11/XF86keysym.h>
 #include <ctype.h>
 #include <locale.h>
 #include <stdio.h>
@@ -500,6 +501,7 @@ insert:
 		if (lines > 0)
 			return;
 		/* fallthrough */
+	case XF86XK_AudioRaiseVolume:
 	case XK_Up:
 	case XK_KP_Up:
 		if (sel && sel->left && (sel = sel->left)->right == curr) {
@@ -521,6 +523,7 @@ insert:
 		sel = curr = prev;
 		calcoffsets();
 		break;
+	case XF86XK_PowerOff:
 	case XK_Return:
 	case XK_KP_Enter:
 		if (!(ev->state & ControlMask)) {
@@ -544,6 +547,7 @@ insert:
 		if (lines > 0)
 			return;
 		/* fallthrough */
+	case XF86XK_AudioLowerVolume:
 	case XK_Down:
 	case XK_KP_Down:
 		if (sel && sel->right && (sel = sel->right) == next) {
-- 
2.35.1

