From 8c413d25d4bca1d2d8a1c1d4149aff5080f864ab Mon Sep 17 00:00:00 2001
From: Arthur Williams <taaparthur@gmail.com>
Date: Sun, 21 Mar 2021 01:23:11 -0700
Subject: [PATCH 7/8] Wrap selection

---
 dmenu.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/dmenu.c b/dmenu.c
index ef4d548..61a504d 100644
--- a/dmenu.c
+++ b/dmenu.c
@@ -464,6 +464,7 @@ insert:
 			return;
 		insert(NULL, nextrune(-1) - cursor);
 		break;
+jump_to_end:
 	case XK_End:
 	case XK_KP_End:
 		if (text[cursor] != '\0') {
@@ -484,6 +485,7 @@ insert:
 	case XK_Escape:
 		cleanup();
 		exit(1);
+jump_to_start:
 	case XK_Home:
 	case XK_KP_Home:
 		if (sel == matches) {
@@ -505,6 +507,9 @@ insert:
 	case XF86XK_AudioRaiseVolume:
 	case XK_Up:
 	case XK_KP_Up:
+		if(sel && !sel->left) {
+			goto jump_to_end;
+		}
 		if (sel && sel->left && (sel = sel->left)->right == curr) {
 			curr = prev;
 			calcoffsets();
@@ -551,6 +556,9 @@ insert:
 	case XF86XK_AudioLowerVolume:
 	case XK_Down:
 	case XK_KP_Down:
+		if(sel && !sel->right) {
+			goto jump_to_start;
+		}
 		if (sel && sel->right && (sel = sel->right) == next) {
 			curr = next;
 			calcoffsets();
-- 
2.35.1

