From 828ce2c63a1538ffab4d369f0d70a16f41eec64c Mon Sep 17 00:00:00 2001
From: Arthur Williams <taaparthur@gmail.com>
Date: Sat, 1 Jan 2022 10:22:12 -0800
Subject: [PATCH 8/8] Exit on mouse press

---
 dmenu.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/dmenu.c b/dmenu.c
index 61a504d..87f854c 100644
--- a/dmenu.c
+++ b/dmenu.c
@@ -342,6 +342,12 @@ movewordedge(int dir)
 	}
 }
 
+static void
+mousepress(XKeyEvent *ev) {
+    cleanup();
+    exit(1);
+}
+
 static void
 keypress(XKeyEvent *ev)
 {
@@ -655,6 +661,9 @@ run(void)
 		case KeyPress:
 			keypress(&ev.xkey);
 			break;
+		case ButtonPress:
+			mousepress(&ev.xkey);
+			break;
 		case SelectionNotify:
 			if (ev.xselection.property == utf8)
 				paste();
@@ -754,7 +763,7 @@ setup(void)
 	/* create menu window */
 	swa.override_redirect = True;
 	swa.background_pixel = scheme[SchemeNorm][ColBg].pixel;
-	swa.event_mask = ExposureMask | KeyPressMask | VisibilityChangeMask;
+	swa.event_mask = ExposureMask | KeyPressMask | ButtonPressMask | VisibilityChangeMask;
 	win = XCreateWindow(dpy, parentwin, x, y, mw, mh, 0,
 	                    CopyFromParent, CopyFromParent, CopyFromParent,
 	                    CWOverrideRedirect | CWBackPixel | CWEventMask, &swa);
-- 
2.35.1

