--- a/config.def.h
+++ b/config.def.h
@@ -42,7 +42,7 @@
 /* number of clients in master area */
 #define NMASTER 1
 /* scroll back buffer size in lines */
-#define SCROLL_HISTORY 500
+#define SCROLL_HISTORY 2000
 /* printf format string for the tag in the status bar */
 #define TAG_SYMBOL   "[%s]"
 /* curses attributes for the currently selected tags */
--- a/config.def.h
+++ b/config.def.h
@@ -78,8 +78,10 @@ static Layout layouts[] = {
 
 /* you can specifiy at most 3 arguments */
 static KeyBinding bindings[] = {
+	{ { CTRL('Y'),         }, { scrollback_line,{ "-1" }                    } },
+	{ { CTRL('E'),         }, { scrollback_line,{ "1"  }                    } },
 	{ { MOD, 'c',          }, { create,         { NULL }                    } },
-	{ { MOD, 'C',          }, { create,         { NULL, NULL, "$CWD" }      } },
+	{ { MOD, 't',          }, { create,         { NULL, NULL, "$CWD" }      } },
 	{ { MOD, 'x', 'x',     }, { killclient,     { NULL }                    } },
 	{ { MOD, 'j',          }, { focusnext,      { NULL }                    } },
 	{ { MOD, 'J',          }, { focusdown,      { NULL }                    } },
@@ -179,7 +183,7 @@ static const ColorRule colorrules[] = {
 # define CONFIG_MOUSE /* compile in mouse support if we build against ncurses */
 #endif
 
-#define ENABLE_MOUSE true /* whether to enable mouse events by default */
+#define ENABLE_MOUSE false /* whether to enable mouse events by default */
 
 #ifdef CONFIG_MOUSE
 static Button buttons[] = {
--- a/dvtm.c
+++ b/dvtm.c
@@ -196,6 +196,7 @@ static void paste(const char *args[]);
 static void quit(const char *args[]);
 static void redraw(const char *args[]);
 static void scrollback(const char *args[]);
+static void scrollback_line(const char *args[]);
 static void send(const char *args[]);
 static void setlayout(const char *args[]);
 static void incnmaster(const char *args[]);
@@ -1340,6 +1341,20 @@ scrollback(const char *args[]) {
 	curs_set(vt_cursor_visible(sel->term));
 }
 
+static void
+scrollback_line(const char *args[]) {
+	if (!is_content_visible(sel))
+		return;
+
+	if (!args[0] || atoi(args[0]) < 0)
+		vt_scroll(sel->term, -1);
+	else
+		vt_scroll(sel->term,  1);
+
+	draw(sel);
+	curs_set(vt_cursor_visible(sel->term));
+}
+
 static void
 send(const char *args[]) {
 	if (sel && args && args[0])
