From ebeb7ba8057d235921ad9d02860bdf9ad8cbf2f4 Mon Sep 17 00:00:00 2001
From: Arthur Williams <taaparthur@gmail.com>
Date: Tue, 21 Apr 2020 11:26:00 -0700
Subject: [PATCH 06/14] Applied clipboard patch

---
 x.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/x.c b/x.c
index 89786b8..4e6f6c8 100644
--- a/x.c
+++ b/x.c
@@ -675,6 +675,8 @@ setsel(char *str, Time t)
 	XSetSelectionOwner(xw.dpy, XA_PRIMARY, xw.win, t);
 	if (XGetSelectionOwner(xw.dpy, XA_PRIMARY) != xw.win)
 		selclear();
+
+	xclipcopy();
 }
 
 void
-- 
2.33.0

