From dc47dba28d4cebdf53399e40139287f6513c56d4 Mon Sep 17 00:00:00 2001
From: Arthur Williams <taaparthur@gmail.com>
Date: Sun, 7 Mar 2021 13:42:19 -0800
Subject: [PATCH 14/14] Added externalpipe keybindings

---
 config.def.h | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/config.def.h b/config.def.h
index 76ceee1..ccbcefb 100644
--- a/config.def.h
+++ b/config.def.h
@@ -16,7 +16,8 @@ static int borderpx = 2;
  * 4: value of shell in /etc/passwd
  * 5: value of shell in config.h
  */
-static char *shell = "/bin/sh";
+#define SHELL "/bin/sh"
+static char *shell = SHELL;
 char *utmp = NULL;
 /* scroll program: to enable use a string like "scroll" */
 char *scroll = NULL;
@@ -178,7 +179,6 @@ static MouseShortcut mshortcuts[] = {
 };
 
 /* Internal keyboard shortcuts. */
-#define MODKEY Mod1Mask
 #define TERMMOD (ControlMask|ShiftMask)
 
 static Shortcut shortcuts[] = {
@@ -197,6 +197,8 @@ static Shortcut shortcuts[] = {
 	{ TERMMOD,              XK_Y,           selpaste,       {.i =  0} },
 	{ ShiftMask,            XK_Insert,      selpaste,       {.i =  0} },
 	{ TERMMOD,              XK_Num_Lock,    numlock,        {.i =  0} },
+	{ TERMMOD,              XK_U,           externalpipe,   {.v =  (char*[]){ SHELL, "-c", "sed -En 's_.*((https?|ftp|gopher|mailto)://\\S*).*$_\\1_p' | uniq | dmenu -w \"$WINDOWID\" | xsel -p", NULL }}},
+	{ TERMMOD,              XK_E,           externalpipe,   {.v =  (char*[]){ SHELL, "-c", "F=\"$(mktemp)\"; trap \"rm $F\" EXIT; cat - >\"$F\"; st vim \"$F\"", NULL }}},
 	{ TERMMOD,              XK_T,           newterm,        {.i =  0} },
 };
 
-- 
2.33.0

