From 57c12b830389577a35f3872857d78fa082937c37 Mon Sep 17 00:00:00 2001
From: Arthur Williams <taaparthur@gmail.com>
Date: Tue, 21 Apr 2020 11:49:14 -0700
Subject: [PATCH 11/15] Changed default font and scroll command

---
 config.def.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config.def.h b/config.def.h
index 6c4cd51..12df972 100644
--- a/config.def.h
+++ b/config.def.h
@@ -5,7 +5,7 @@
  *
  * font: see http://freedesktop.org/software/fontconfig/fontconfig-user.html
  */
-static char *font = "Liberation Mono:pixelsize=12:antialias=true:autohint=true";
+static char *font = "Ubuntu Mono:size=16";
 static int borderpx = 2;
 
 /*
-- 
2.30.1

