From e1480cc5ab418e182abbfd5b0a493eec1cd6bc9c Mon Sep 17 00:00:00 2001
From: Arthur Williams <taaparthur@gmail.com>
Date: Mon, 2 Nov 2020 15:56:12 -0800
Subject: [PATCH] Reduce size of makefile

---
 Makefile  | 23 +++--------------------
 config.mk |  5 +----
 smdev.c   |  1 +
 util.h    |  2 +-
 4 files changed, 6 insertions(+), 25 deletions(-)

diff --git a/Makefile b/Makefile
index b79e3d8..0e1b994 100644
--- a/Makefile
+++ b/Makefile
@@ -19,7 +19,7 @@ OBJ = $(SRC:.c=.o) $(LIB)
 BIN = $(SRC:.c=)
 MAN = $(SRC:.c=.1)
 
-all: options binlib
+all: options $(BIN)
 
 options:
 	@echo mdev build options:
@@ -27,30 +27,13 @@ options:
 	@echo "LDFLAGS  = $(LDFLAGS)"
 	@echo "CC       = $(CC)"
 
-binlib: util.a
-	$(MAKE) bin
-
-bin: $(BIN)
-
-$(OBJ): config.h util.h config.mk
+$(BIN): config.h util.h $(OBJ)
+	${CC} ${CFLAGS} $^ -o $@ ${LDFLAGS}
 
 config.h:
 	@echo creating $@ from config.def.h
 	@cp config.def.h $@
 
-.o:
-	@echo LD $@
-	@$(LD) -o $@ $< util.a $(LDFLAGS)
-
-.c.o:
-	@echo CC $<
-	@$(CC) -c -o $@ $< $(CFLAGS)
-
-util.a: $(LIB)
-	@echo AR $@
-	@$(AR) -r -c $@ $(LIB)
-	@ranlib $@
-
 install: all
 	@echo installing executable to $(DESTDIR)$(PREFIX)/bin
 	@mkdir -p $(DESTDIR)$(PREFIX)/bin
diff --git a/config.mk b/config.mk
index e8a717c..cd6d931 100644
--- a/config.mk
+++ b/config.mk
@@ -5,7 +5,4 @@ VERSION = 0.2.3
 PREFIX = /usr/local
 
 CC = cc
-LD = $(CC)
-CPPFLAGS = -D_DEFAULT_SOURCE -Wno-format-truncation
-CFLAGS   = -std=c99 -Wall -pedantic $(CPPFLAGS)
-LDFLAGS  = -s
+CFLAGS   = -std=c99 -Wall -pedantic -D_DEFAULT_SOURCE -Wno-format-truncation
diff --git a/smdev.c b/smdev.c
index 7f89be2..a7e3767 100644
--- a/smdev.c
+++ b/smdev.c
@@ -22,6 +22,7 @@
 #include <string.h>
 #include <unistd.h>
 
+#include "arg.h"
 #include "config.h"
 #include "mkpath.h"
 #include "util.h"
diff --git a/util.h b/util.h
index d1b0e8c..b1d31fa 100644
--- a/util.h
+++ b/util.h
@@ -1,5 +1,5 @@
 /* See LICENSE file for copyright and license details. */
-#include "arg.h"
+#include "stddef.h"
 
 #define LEN(x) (sizeof (x) / sizeof *(x))
 
-- 
2.28.0

