From b420cbf563f9647dfaeacb6039b37d4b7a651065 Mon Sep 17 00:00:00 2001
From: Arthur Williams <taaparthur@disroot.org>
Date: Fri, 24 Dec 2021 17:22:54 -0600
Subject: [PATCH 1/3] map jj to escape

---
 vi.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/vi.c b/vi.c
index 432c713..b54b8a2 100644
--- a/vi.c
+++ b/vi.c
@@ -195,6 +195,7 @@ static struct macro_state macro;
 enum expand_mode { NONE, EXPAND, COMPLETE, PRINT };
 static enum expand_mode expanded = NONE;/* last input was expanded */
 
+static int last_char;
 int
 x_vi(char *buf, size_t len)
 {
@@ -237,8 +238,14 @@ x_vi(char *buf, size_t len)
 				continue;
 			}
 		}
+        if(insert && (last_char == 'j' && c == 'j' || last_char == 'k' && c == 'k')) {
+            vi_hook(edchars.erase);
+            c = CTRL('[');
+        }
+
 		if (vi_hook(c))
 			break;
+        last_char = c;
 		x_flush();
 	}
 
-- 
2.33.0

