From cba17db1b5366491f7e4a589a1c04c5b840c7b9e Mon Sep 17 00:00:00 2001
From: Arthur Williams <taaparthur@disroot.org>
Date: Sat, 25 Dec 2021 22:13:57 -0600
Subject: [PATCH 4/4] Escape spaces during auto completion

---
 edit.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/edit.c b/edit.c
index 8917f5f..2788398 100644
--- a/edit.c
+++ b/edit.c
@@ -910,7 +910,7 @@ x_escape(const char *s, size_t len, int (*putbuf_func) (const char *, size_t))
 	int rval = 0;
 
 	for (add = 0, wlen = len; wlen - add > 0; add++) {
-		if (strchr("!\"#$&'()*:;<=>?[\\]`{|}", s[add]) ||
+		if (strchr("!\"#$&'()*:;<=>?[\\]`{|} ", s[add]) ||
 		    strchr(ifs, s[add])) {
 			if (putbuf_func(s, add) != 0) {
 				rval = -1;
-- 
2.33.0

