From 02756b04d45df7c7d686794843bd07e68cab6074 Mon Sep 17 00:00:00 2001
From: Arthur Williams <taaparthur@disroot.org>
Date: Sat, 11 Dec 2021 00:20:02 -0800
Subject: [PATCH 3/4] Change bindings when fullscreen

---
 mtm.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/mtm.c b/mtm.c
index 41b9a82..0c5279a 100644
--- a/mtm.c
+++ b/mtm.c
@@ -1117,10 +1117,10 @@ handlechar(int r, int k) /* Handle a single input character. */
     DO(false, CODE(KEY_F(10)),     SEND(n, "\033[21~"); SB)
     DO(false, CODE(KEY_F(11)),     SEND(n, "\033[23~"); SB)
     DO(false, CODE(KEY_F(12)),     SEND(n, "\033[24~"); SB)
-    DO(true,  MOVE_UP,             focus(findnode(root, ABOVE(n))))
-    DO(true,  MOVE_DOWN,           focus(findnode(root, BELOW(n))))
-    DO(true,  MOVE_LEFT,           focus(findnode(root, LEFT(n))))
-    DO(true,  MOVE_RIGHT,          focus(findnode(root, RIGHT(n))))
+    DO(true,  MOVE_UP,             focus(!isfullscreen ? findnode(root, ABOVE(n)) : BACKWARD(n)))
+    DO(true,  MOVE_DOWN,           focus(!isfullscreen ? findnode(root, BELOW(n)) : FORWARD (n)))
+    DO(true,  MOVE_LEFT,           focus(!isfullscreen ? findnode(root, LEFT (n)) : BACKWARD(n)))
+    DO(true,  MOVE_RIGHT,          focus(!isfullscreen ? findnode(root, RIGHT(n)) : FORWARD (n)))
     DO(true,  MOVE_OTHER,          focus(lastfocused))
     DO(true,  MOVE_BACKWARD,       focus(BACKWARD(n)))
     DO(true,  MOVE_FORWARD,        focus(FORWARD(n)))
-- 
2.33.0

