diff --git a/window.c b/window.c
index 6f9a3906..eb534a70 100644
--- a/window.c
+++ b/window.c
@@ -154,6 +154,7 @@ void win_open(win_t *win)
 	Pixmap none;
 	int gmask;
 	XSizeHints sizehints;
+	XWMHints hints;
 
 	e = &win->env;
 	parent = options->embed != 0 ? options->embed : RootWindow(e->dpy, e->scr);
@@ -252,6 +253,11 @@ void win_open(win_t *win)
 	sizehints.y = win->y;
 	XSetWMNormalHints(win->env.dpy, win->xwin, &sizehints);
 
+	hints.flags = InputHint | StateHint;
+	hints.input = 1;
+	hints.initial_state = NormalState;
+	XSetWMHints(win->env.dpy, win->xwin, &hints);
+
 	win->h -= win->bar.h;
 
 	win->buf.w = e->scrw;
